src/core/reference/src/op/nms_rotated.cpp

```diff
@@ -195,7 +195,12 @@ void nms_rotated(const float* boxes_data,
     }
 
     if (sort_result_descending) {
-        std::reverse(filteredBoxes.begin(), filteredBoxes.end());
+        std::stable_sort(filteredBoxes.begin(), filteredBoxes.end(), [](const BoxInfo& l, const BoxInfo& r) {
+            return (l.score > r.score) || (l.score == r.score && l.batch_index < r.batch_index) ||
+                   (l.score == r.score && l.batch_index == r.batch_index && l.class_index < r.class_index) ||
+                   (l.score == r.score && l.batch_index == r.batch_index && l.class_index == r.class_index &&
+                    l.index < r.index);
```

Comment:
```suggestion
            return (l.score > r.score) || ((l.score == r.score) && (std::tie(l.batch_index, l.class_index, l.index) < std::tie(r.batch_index, r.class_index, r.index)));
```
Will it give same result?
Why `score` member is compared using greater other members using less?

Reply:
Update to use `std::tie`, thanks! As I understand the goal is to sort descending by score, but ascending for batch and class.
It is aligned with CPU:

https://github.com/openvinotoolkit/openvino/blob/bc463e886b21c27fbce0293c5a5fa313a0eff0c0/src/plugins/intel_cpu/src/nodes/non_max_suppression.cpp#L788-L796